### PR TITLE
Starting tasks

### LINKED ISSUE #7103: Social-Themed Starting Tasks
Now that both mobile apps have Guild access, we'd like to make some extra starting tasks that show up in new accounts. We hope that these will help encourage new Habiticans to become members of the community!

Alternate task and wording suggestions are very welcome!

These To-Dos should appear underneath the "Join Habitica" To-Do. The tasks on the website should be different from the ones on the mobile app, because the directions to get to the pages will be different. Of course, if the user tries both platforms the tasks will be incorrect one of them, but I am not sure there is an easy way to change that. Someone please chime in if I am wrong!

WEB

Learn about Habitica
Checklist: 
1) Read the Overview for New Users (on the top bar, under Help)
2) Read the FAQ (under Help)

Join the Newbies Guild (under Social > Guilds)
Note: This is the best place to ask questions!

Introduce yourself in the Tavern (Social > Tavern Chat)
Checklist:
1) Read the Community Guidelines
2) Post a fun fact about yourself, like a habit you want to improve or your favorite hobby!

MOBILE

Read the Habitica FAQ 
Note: Under Menu > About > Help & FAQ

Join the Newbies Guild (Menu > Social > Guilds)
Note: Ask questions here!

Introduce yourself in the Tavern (Menu > Social > Tavern Chat)
Checklist:
1) Read the Community Guidelines
2) Post a fun fact about yourself, like a habit you want to improve or your favorite hobby!

### PR DESCRIPTION
**Pull request by librarianmage**
Fixes #7103 
### Changes

Adds social-themed tasks for new users:
screenshot 2016-05-30 at 6 10 55 pm

---

UUID:

### CODE REVIEW

**Comment by librarianmage** [comment]
@crookedneighbor The tests seem to be running with npm v2:
screenshot 2016-05-31 at 11 01 39 am
Could that be why my build is failing?

**Comment by crookedneighbor** [comment]
It's not using npm v2. v2 is what comes standard with node v4, but if you look at line 161, you'll see that `npm install -g npm@3` gets run which install npm 3.

**Comment by librarianmage** [comment]
@crookedneighbor Ok. I'm just not sure where the errors come from, and if they are because of my changes.

File: common/locales/en/defaultTasks.json [hunk 1]
```diff
@@ -8,10 +8,24 @@
...
+
+  "defaultTodo3Text": "Join the Newbies Guild",
+  "defaultTodo3Notes": "☰ > Social > Guilds/n/nThis is the best place to ask questions!",
+
+  "defaultTodo4Text": "Introduce Yourself to the Tavern",
+  "defaultTodo4Notes": "☰ > Social > Tavern Chat",
+  "defaultTodo4Checklist1": "Read the Community Guidelines",
+  "defaultTodo4Checklist2": "Post a fun fact about yourself, like a habit you want to improve or your favorite hobby!",
 
+
```
Comment (crookedneighbor) [inline common/locales/en/defaultTasks.json]: Why this extra space?

**Comment by paglias** [comment]
There's a problem in models/user.js#584. It should return a new object for the checklist instead of modifying the existing one

**Comment by lemoness** [comment]
Is there a way to have the task "Read Overview for New Users" simply
display only for website users, rather than have it say "website only" in
the task itself? If not, that's fine.

**Comment by librarianmage** [comment]
I don't believe we can, unfortunately.

File: common/locales/en/defaultTasks.json [hunk 2]
```diff
@@ -8,10 +8,24 @@
   "defaultHabit3Text": "Take the Stairs/Elevator (Click the pencil to edit)",
   "defaultHabit3Notes": "Sample Good or Bad Habits: +/- Took Stairs/Elevator ; +/- Drank Water/Soda",
 
-  "defaultTodoNotes": "You can either complete this To-Do, edit it, or remove it.",
-
+
```
Comment (awitherow) [inline common/locales/en/defaultTasks.json]: remove space?

File: common/locales/en/defaultTasks.json (hunk 1 above)
Comment (librarianmage) [inline common/locales/en/defaultTasks.json]: No idea, I probably wanted to space out the habits, todos, and rewards

**Comment by librarianmage** [comment]
@paglias is there something I should change?

**Comment by MathWhiz** [comment]
@paglias would
```javascript
return _.map(taskDefaults.checklist, (checklistItem) => {
  checklistItem.text = checklistItem.text(user.preferences.language);
  return checklistItem;
```
work?

**Comment by librarianmage** [comment]
@crookedneighbor Would you be able to help me with the tests?

**Comment by librarianmage** [comment]
@crookedneighbor What went wrong?

File: test/api/v3/integration/tasks/GET-tasks_user.test.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ describe('GET /tasks/user', () => {
   it('returns all user\'s tasks', async () => {
     let createdTasks = await user.post('/tasks/user', [{text: 'test habit', type: 'habit'}, {text: 'test todo', type: 'todo'}]);
     let tasks = await user.get('/tasks/user');
-    expect(tasks.length).to.equal(createdTasks.length + 1); // + 1 because 1 is a default task
+    expect(tasks.length).to.equal(createdTasks.length + 4); // + 1 because 1 is a default task
```
Comment (khipkin) [inline test/api/v3/integration/tasks/GET-tasks_user.test.js]: Suggestion: update comment too

File: test/api/v3/integration/user/GET-user_anonymized.test.js [hunk 4]
```diff
@@ -75,7 +75,7 @@ describe('GET /user/anonymized', () => {
     });
     // tasks
     expect(tasks2).to.exist;
-    expect(tasks2.length).to.eql(5); // +1 because generateUser() assigns one todo
+    expect(tasks2.length).to.eql(8); // +1 because generateUser() assigns one todo
```
Comment (khipkin) [inline test/api/v3/integration/user/GET-user_anonymized.test.js]: Suggestion: here as well?

File: test/api/v3/integration/dataexport/GET-export_userdata.json.test.js [hunk 5]
```diff
@@ -23,7 +23,7 @@ describe('GET /export/userdata.json', () => {
     expect(res.tasks.dailys[0]._id).to.equal(tasks[1]._id);
     expect(res.tasks.rewards.length).to.equal(1);
     expect(res.tasks.rewards[0]._id).to.equal(tasks[2]._id);
-    expect(res.tasks.todos.length).to.equal(2);
+    expect(res.tasks.todos.length).to.equal(5);
     expect(res.tasks.todos[1]._id).to.equal(tasks[3]._id);
```
Comment (khipkin) [inline test/api/v3/integration/dataexport/GET-export_userdata.json.test.js]: [This test failure](https://travis-ci.org/HabitRPG/habitrpg/builds/151077184#L9979) is due to this assertion.

File: test/api/v3/integration/dataexport/GET-export_userdata.xml.test.js [hunk 6]
```diff
@@ -36,7 +36,7 @@ describe('GET /export/userdata.xml', () => {
     expect(res.tasks.dailys[0]._id).to.equal(tasks[1]._id);
     expect(res.tasks.rewards.length).to.equal(2);
     expect(res.tasks.rewards[0]._id).to.equal(tasks[2]._id);
-    expect(res.tasks.todos.length).to.equal(3);
+    expect(res.tasks.todos.length).to.equal(6);
     expect(res.tasks.todos[1]._id).to.equal(tasks[3]._id);
```
Comment (khipkin) [inline test/api/v3/integration/dataexport/GET-export_userdata.xml.test.js]: [This test failure](https://travis-ci.org/HabitRPG/habitrpg/builds/151077184#L10000) is due to this assertion.

**Comment by khipkin** [comment]
@MathWhiz The tests I commented on make assumptions about tasks' specific indices in `res.tasks.todos` that are no longer true.  I think the problems are just that the indices are not correct.

**Comment by crookedneighbor** [comment]
I think the generateUser function should automatically remove any tasks that are added as a result of registering. Some tests may need to be updated, but it will eliminate these weird cases in the future.

**Comment by librarianmage** [comment]
@crookedneighbor Would an option to keep the registered tasks in the `generateUser()` function be a good idea?

**Comment by crookedneighbor** [comment]
IMO, I think the user should just be blank. No tasks at all.

**Comment by librarianmage** [comment]
The problem with that is that it means I can't (easily) make a test to
check if the default tasks are added.

**Comment by crookedneighbor** [comment]
No, you would use the register route instead to test that.

**Comment by MathWhiz** [comment]
Ok

File: test/helpers/api-integration/v3/object-generators.js [hunk 7]
```diff
@@ -28,6 +30,8 @@ export async function generateUser (update = {}) {
 
   let apiUser = new ApiUser(user);
 
+  await apiUser.post('/user/reset');
```
Comment (crookedneighbor) [inline test/helpers/api-integration/v3/object-generators.js:33]: Clever 😄

File: common/locales/en/defaultTasks.json [hunk 8]
```diff
@@ -11,6 +11,20 @@
...
   "defaultTodo1Text": "Join Habitica (Check me off!)",
+  "defaultTodo1Notes": "You can either complete this To-Do, edit it, or remove it.",
+
+  "defaultTodo2Text": "Learn About Habitica",
+  "defaultTodo2Notes": "Overview of New Users:\n\n💻 only: ☰(Menu - Top Bar) > Help\n\nFAQ:\n\n📱: ☰ > Help & FAQ\n\n💻 : ☰ > Help > FAQ",
+  "defaultTodo2Checklist1": "Read the Overview of New Users (Website only)",
+  "defaultTodo2Checklist2": "Read the FAQ",
+
+  "defaultTodo3Text": "Join the Newbies Guild",
+  "defaultTodo3Notes": "☰ > Social > Guilds/n/nThis is the best place to ask questions!",
```
Comment (crookedneighbor) [inline common/locales/en/defaultTasks.json]: Your `/n` should be `\n`

screen shot 2016-09-12 at 8 08 20 am

File: common/locales/en/defaultTasks.json [hunk 9]
```diff
@@ -11,6 +11,20 @@
   "defaultTodoNotes": "You can either complete this To-Do, edit it, or remove it.",
 
   "defaultTodo1Text": "Join Habitica (Check me off!)",
+  "defaultTodo1Notes": "You can either complete this To-Do, edit it, or remove it.",
+
+  "defaultTodo2Text": "Learn About Habitica",
+  "defaultTodo2Notes": "Overview of New Users:\n\n💻 only: ☰(Menu - Top Bar) > Help\n\nFAQ:\n\n📱: ☰ > Help & FAQ\n\n💻 : ☰ > Help > FAQ",
```
Comment (crookedneighbor) [inline common/locales/en/defaultTasks.json]: Personally, I don't like this. Looks messy.

screen shot 2016-09-12 at 8 09 44 am

Comment (crookedneighbor) [inline common/locales/en/defaultTasks.json]: Any other maintainers have thoughts on that?

Comment (librarianmage) [reply common/locales/en/defaultTasks.json]: What do you think I should do?

Comment (librarianmage) [reply common/locales/en/defaultTasks.json]: @crookedneighbor How about putting the info in the checklist?

Comment (crookedneighbor) [reply common/locales/en/defaultTasks.json]: That's not my call to make. I just don't think this current version looks right.

Comment (librarianmage) [reply common/locales/en/defaultTasks.json]: I can try splitting it into two different tasks...

File: website/common/locales/en/defaultTasks.json [hunk 10]
```diff
@@ -11,6 +11,20 @@
   "defaultTodoNotes": "You can either complete this To-Do, edit it, or remove it.",
 
   "defaultTodo1Text": "Join Habitica (Check me off!)",
+  "defaultTodo1Notes": "You can either complete this To-Do, edit it, or remove it.",
+
+  "defaultTodo2Text": "Learn About Habitica",
+  "defaultTodo2Notes": "Overview of New Users:\n\n💻 only: ☰(Menu - Top Bar) > Help\n\nFAQ:\n\n📱: ☰ > Help & FAQ\n\n💻 : ☰ > Help > FAQ",
```
Comment (crookedneighbor) [inline website/common/locales/en/defaultTasks.json:17]: @lemoness or @alys thoughts on this text?

**Review by crookedneighbor — CHANGES_REQUESTED** [review]

File: website/common/locales/en/defaultTasks.json (hunk 10 above)
Comment (Alys) [reply website/common/locales/en/defaultTasks.json:17]: @MathWhiz Could you please show us a screenshot of what `defaultTodo2Notes` looks like when you hover over the task?

Comment (librarianmage) [reply website/common/locales/en/defaultTasks.json:17]: @Alys it currently looks like:

img

**Comment by lemoness** [comment]
I think it's a little unclear -- people may not necessarily understand what
the instructions are telling them to do.

**Comment by librarianmage** [comment]
@lemoness How about splitting it into two tasks? Something like:
- See Overview of New Users (Computer Only)
- See FAQ

**Comment by lemoness** [comment]
I think that could work! Or maybe have it say, "Read FAQ" etc, since that's
a little more specific about what they should do. (I assume that "Computer
Only" is not in the task name, but rather a sign of where it would appear.)

**Comment by librarianmage** [comment]
@lemoness Unfortunately, having the Overview of New Users on only be shown on the computer is not possible (to my knowledge)

**Comment by crookedneighbor** [comment]
It is possible to specify what the default tasks are based on what platform was used for sign up. Pretty sure we already do that.

**Comment by librarianmage** [comment]
@crookedneighbor What about people that initially sign up on one device, and use both? If someone uses the website at all, it would be useful for them to read the Overview of New Users.

**Comment by lemoness** [comment]
For now, let's just focus on what device they used to sign up. That way, we
can get the changes out sooner.

**Comment by librarianmage** [comment]
@crookedneighbor I don't see anything in [auth.js](https://github.com/HabitRPG/habitica/blob/develop/website/server/controllers/api-v3/auth.js) that would suggest that there is a difference in how mobile and non-moblie users register, but I'm not sure what I'd look for.

**Comment by crookedneighbor** [comment]
https://github.com/HabitRPG/habitica/blob/develop/website/server/models/user/hooks.js

**Comment by librarianmage** [comment]
@crookedneighbor How do you suggest I change the logic? It is currently checking for the web/android or ios, and I want specific tasks for web and android/ios

**Comment by librarianmage** [comment]
@crookedneighbor I don't think I will be able to finish this. The logic behind setting up tasks is a tad too complex for me to change easily (AFAIK), and I've been stuck for the past week. I'm sorry. 😞

**Comment by Alys** [comment]
@MathWhiz Don't be sorry! Thank you for trying, and thanks very much for letting us know! :heart:

**Pull request closed without merging**